packages/lit-html/src/lib/lit-html.ts

```diff
@@ -888,6 +917,58 @@ export class EventPart extends AttributePart {
   }
 }
 
+export class SpreadPart {
+  __value: unknown;
+  __directive?: Directive;
+
+  constructor(public __element: Element) {}
+
+  __setValue(value: unknown): void {
+    const directive = (value as DirectiveResult)._$litDirective$;
+    if (directive !== undefined) {
+      if (this.__directive?.constructor !== directive) {
+        this.__directive = new directive(this);
+        // TODO (justinfagnani): clear?
+      }
+      value = this.__directive.update(this, (value as DirectiveResult).values);
+    }
+    if (value === nothing) {
+      // ?
+      return;
+    }
+    if ((value as TemplateResult)._$litType$ !== ATTR_RESULT) {
+      throw new Error('only Attributes can be set here');
+    }
+    const {strings, values} = value as TemplateResult;
+    const element = this.__element;
+    let from = 0;
+    let template = attributesTemplateCache.get(strings);
+    if (template === undefined) {
+      attributesTemplateCache.set(
+        strings,
+        (template = new AttributesTemplate(strings))
+      );
+    }
+    for (const {name, strings} of template.__attrs) {
+      if (strings.length === 1) {
+        // TODO: static, only set once
+        element.setAttribute(name, strings[0]);
+      } else {
+        let value = strings[0];
+        const size = strings.length - 1;
+        for (let i = 0; i < size; i++) {
+          const v = values[from++];
+          value +=
+            (typeof v === 'string' ? v : v == null ? '' : String(v)) +
+            strings[i + 1];
+        }
+        // TODO: dirty-check
+        element.setAttribute(name, value as string);
```

Comment:
What about property and event bindings and values that are not strings? This feature doesn't seem worth it unless you can bind what you normally can in attribute position.

Reply:
N/A